quinn/src/recv_stream.rs

```diff
@@ -21,8 +21,40 @@ use crate::{
 /// - A variant of [`ReadError`] has been yielded by a read call
 /// - [`stop()`] was called explicitly
 ///
+/// # Closing a stream
+///
+/// When a stream is expected to be closed gracefully the sender should call
+/// [`SendStream::finish`].  However there is no guarantee the connected [`RecvStream`] will
+/// receive this End Of File (EOF) indicator in the same QUIC frame as the last frame which
+/// carried data.
+///
+/// So even if the application layer logic already knows it read all the data because it
+/// does its own framing, it must still read from the [`RecvStream`] until EOF.  Otherwise
```

Comment:
```suggestion
/// does its own framing, it must still read from the [`RecvStream`] until finished.  Otherwise
```

Reply:
Alternatively, maybe ``read to the end of the [`RecvStream`]``? That's more consistent with the method name, while still avoiding introduction of inconsistent terms.